submissions/Oleksii_Drohachov/Frogger_task/app.js

```diff
@@ -0,0 +1,182 @@
+
+
+const Entity = function (xAxisPosition, yAxisPosition, sprite) {
+  this.xAxisPosition = xAxisPosition;
+  this.yAxisPosition = yAxisPosition;
+  this.width = entityWidth;
+  this.height = entityHeight;
+  this.sprite = sprite;
+}
+
+const Player = function (xAxisPosition, yAxisPosition, sprite, width, height) {
+  Entity.call(this, xAxisPosition, yAxisPosition, sprite, width, height)
+  this.moveXAxis = tileWidth;
+  this.moveYAxis = tileHeight;
+};
+
+Player.prototype = Object.create(Entity.prototype);
+
+Player.prototype.update = function () {
+  if (player.yAxisPosition < 0) {
+    this.respawn();
+    scoreBoard.update();
+  }
+};
+
+Player.prototype.render = function () {
+  ctx.drawImage(
+    Resources.get(this.sprite),
+    this.xAxisPosition,
+    this.yAxisPosition
+  );
+};
+
+Player.prototype.respawn = function () {
+  this.xAxisPosition = playerStartingPositionX;
+  this.yAxisPosition = playerStartingPositionY;
+};
+
+Player.prototype.handleInput = function (key) {
+  switch (key) {
+    case "up":
+      this.yAxisPosition -= this.moveYAxis;
+      break;
+    case "down":
+      if (this.yAxisPosition + this.moveYAxis <= fieldHeight - fieldBottomGap) {
+        this.yAxisPosition += this.moveYAxis;
+      }
+      break;
+    case "left":
+      if (this.xAxisPosition - this.moveXAxis >= 0) {
+        this.xAxisPosition -= this.moveXAxis;
+      }
+      break;
+    case "right":
+      if (this.xAxisPosition + this.moveXAxis < fieldWidth) {
+        this.xAxisPosition += this.moveXAxis;
+      }
+      break;
+  }
+};
+
+const player = new Player(playerStartingPositionX, playerStartingPositionY, "images/char-boy.png");
+
+const Enemy = function (xAxisPosition, yAxisPosition, sprite, moveSpeed, width, height, player) {
+  Entity.call(this, xAxisPosition, yAxisPosition, sprite, width, height)
+  this.moveSpeed = moveSpeed;
+  this.player = player;
+};
+
+Enemy.prototype = Object.create(Entity.prototype);
+
+Enemy.prototype.update = function (dt) {
+  this.xAxisPosition += this.moveSpeed * dt;
+
+  if (this.xAxisPosition >= fieldWidth) {
+    this.xAxisPosition = -this.width;
+  }
+
+  this.checkCollision();
+};
+
+Enemy.prototype.render = function () {
+  ctx.drawImage(
+    Resources.get(this.sprite),
+    this.xAxisPosition,
+    this.yAxisPosition
+  );
+};
+
+Enemy.prototype.checkCollision = function () {
+
+  if (
+    player.xAxisPosition < this.xAxisPosition + this.width &&
+    player.width + player.xAxisPosition > this.xAxisPosition &&
+    player.yAxisPosition < this.yAxisPosition + this.height &&
+    player.yAxisPosition + player.height > this.yAxisPosition
+  ) {
+    
+    player.respawn();
+    scoreBoard.score = 0;
+    scoreTitle.textContent = `Current score is: ${scoreBoard.score}`;
+  }
+};
+
+const enemyStats = [
+  {
+    yAxisPosition: firstEnemyRow,
+    moveSpeed: 180,
+  },
+  {
+    yAxisPosition: secondEnemyRow,
+    moveSpeed: 150,
+  },
+  {
+    yAxisPosition: thirdEnemyRow,
```

Comment:
`firstAnythin`, `secondAnything` etc just like `anything1`, `anything2` clearly calls for an array.
`whatever[1]` or `whatever[natural number - 1]` looks more comprehensible.

Also using arrays will enable `Array` mehods. Imagine you are tasked to add more rows for enemies.